app/src/main/java/org/ole/planet/myplanet/ui/viewer/ImageViewerActivity.kt

```diff
@@ -24,15 +25,11 @@ class ImageViewerActivity : AppCompatActivity() {
         val isFullPath = intent.getBooleanExtra("isFullPath", false)
         val imageOpenIntent = intent
         fileName = imageOpenIntent.getStringExtra("TOUCHED_FILE")
-        if (fileName != null && fileName?.isNotEmpty() == true) {
-            val regex = Regex(".+/([^/]+\\.(jpg|jpeg|png|gif|bmp))")
+        val nameWithExtension = FileUtils.extractFileName(fileName)
+        val nameWithoutExtension = nameWithExtension?.substringBefore(".")
+        activityImageViewerBinding.imageFileName.text = nameWithoutExtension
```

Comment:
we can have it example

//FileUtils
fun nameWithoutExtension(fileName: String?): String {
       //code for extraction
        return extractedFileWithoutExtension
    }
    
   this will then be called 
   activityImageViewerBinding.imageFileName.text = nameWithoutExtension(fileName)
   where we are passing the filename to the function for extraction